File: javascript/leap/leap.js
// isLeap returns true/false if a year is a leap year
const isLeap = function (year) {
    if (year < 0) {
        return false;
    }

    if (year % 100 === 0 && year % 400 === 0)
        return true;
    else return year % 4 === 0 && year % 100 !== 0 && year % 400 !== 0;

}

export {isLeap};

File: go/leap/leap.go
// Package leap returns if a year is a leap year
package leap

// IsLeapYear returns true/false if a year is a leap year
func IsLeapYear(year int) bool {
	if year < 0 {
		return false
	}
	switch {
	case year%100 == 0 && year%400 == 0:
		return true
	case year%4 == 0 && year%100 != 0 && year%400 != 0:
		return true
	default:
		return false

	}

}
